enterprise-ws-spec/src/main/asciidoc/ServerProgrammingModel.adoc

```diff
@@ -351,295 +344,180 @@ web.xml or a partial web.xml:
-_javax.ejb.Stateless_ annotation or denoted in the deployment descriptor
+_jakarta.ejb.Stateless_ annotation or denoted in the deployment descriptor
 as a stateless session bean. The bean class no longer implements the
-_javax.ejb.SessionBean_ interface.
+_jakarta.ejb.SessionBean_ interface.
 
 The full requirements for Stateless Session Bean are defined in the
-Enterprise JavaBeans specification (EJB Core Contracts and Requirements)
-section 4.1
+Jakarta Enterprise Beans 4.0 specification (EJB Core Contracts and Requirements).
 
 ====== Allowed access to container services
 
-The Enterprise JavaBeans specification (EJB Core Contracts and
-Requirements) section 4.4.1 defines the allowed container service access
-requirements.
-
-A stateless session bean that implements a web service endpoint using
-the JAX-RPC APIs access the JAX-RPC javax.xml.rpc.handler.MessageContext
-interface by means of the SessionContext.getMessageContext method. Refer
-to section 4.3.6 of the Enterprise JavaBeans specification (EJB Core
-Contracts and Requirements)
+The Jakarta Enterprise Beans 4.0 specification (EJB Core Contracts and
+Requirements) defines the allowed container service access requirements.
 
 A stateless or singleton session bean that implements a web service
-endpoint using the JAX-WS APIs should use the
-javax.xml.ws.WebServiceContext, which can be injected by use of the
-@Resource annotation (see section 2.2 of JSR-250 Common Annotations for
-Java Platform specification), to access message context and security
-information relative to the request being served. The WebServiceContext
-interface allows the stateless or singleton session bean instance to get
-access to the javax.xml.ws.handler.MessageContext. Usage of a
-WebServiceContext must meet the requirements defined by the JAX-WS
-specification section 5.3.
-
-====== javax.ejb.Singleton annotation
+endpoint using the Jakarta XML Web Services APIs should use the
+jakarta.xml.ws.WebServiceContext, which can be injected by use of the
+@Resource annotation (see Jakarta Annotations specification), to access 
+message context and security information relative to the request being 
+served. The WebServiceContext interface allows the stateless or singleton 
+session bean instance to get access to the jakarta.xml.ws.handler.MessageContext. 
+Usage of a WebServiceContext must meet the requirements defined by the 
+Jakarta XML Web Services specification.
+
+====== jakarta.ejb.Singleton annotation
 
 EJB 3.1 introduces Singleton session bean component that provides an
 easy access to shared state. A Singleton session bean is instantiated
 once per application. A Singleton session bean must be annotated with
-the _javax.ejb.Singleton_ annotation or denoted in the deployment
+the _jakarta.ejb.Singleton_ annotation or denoted in the deployment
 descriptor as a singleton session bean.
 
 The full requirements for Singleton Session Bean are defined in the
 section 4.8 of EJB 3.1 specification.
```

Comment:
This should be updated, we're far after EJB 3.1. What about sth like:
"Singleton session bean component, as defined by Enterprise Java Beans specification, provides an
easy access to shared state. ..."?

Reply:
will update as suggested